### PR TITLE
ASUPF-239: Change adapter application interface

### PR DESCRIPTION
**Pull request by Ruitjes**
Gives a user more control over the index

### CODE REVIEW

**Comment by Ruitjes** [comment]
I'm wondering if this version change is a Major or Minor.
Because the API does change and isn't directly (needs small changes) compatible with version 1.0.4. 

What do you think?

Before:
```js
createIndex(data, options) 
const searchClient = getSearchClient();
```

Now:
```js
const index = createIndex(data, options) 
const searchClient = getSearchClient(index);
```

**Comment by Jonne** [comment]
> I'm wondering if this version change is a Major or Minor. Because the API does change and isn't directly (needs small changes) compatible with version 1.0.4.
> 
> What do you think?
> 
> Before:
> 
> ```js
> createIndex(data, options) 
> const searchClient = getSearchClient();
> ```
> 
> Now:
> 
> ```js
> const index = createIndex(data, options) 
> const searchClient = getSearchClient(index);
> ```

We could make the change backwards compatible? So you use both syntax? In that case we dont have to change the major version. In the readme I would only mention the new way though

File: src/adapter.ts [hunk 1]
```diff
@@ -7,7 +7,13 @@ import {
 } from "@algolia/client-search";
 import { ItemsJsOptions, SearchClient } from "./itemsjsInterface";
 
-export function getSearchClient(index: any): SearchClient {
+let index;
+
+export function getSearchClient(newIndex?: any): SearchClient {
+  // This newIndex provides the possitblity to not use the latest created index.
+  // Added to make it backwords compatiable with v1.0.4
+  if (newIndex) index = newIndex;
```
Comment (Jonne) [inline src/adapter.ts:15]: Ik vind deze wat moeilijk te lezen, kun je niet performSearch(newIndex || index) gebruiken?

**Review by Jonne — CHANGES_REQUESTED** [review]

File: src/adapter.ts (hunk 1 above)
Comment (Ruitjes) [reply src/adapter.ts:15]: Didn't know this was possible, indeed an easier fix!

**Review by Jonne — APPROVED** [review]

**Pull request merged**